eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -293,7 +293,10 @@ object Validation {
       } else if (pc.getChannelUpdateSameSideAs(u).exists(_.timestamp >= u.timestamp)) {
         log.debug("ignoring {} (duplicate)", u)
         sendDecision(origins, GossipDecision.Duplicate(u))
-        d
+        // NB: we update the graph because the balances may have changed even if the channel_update is the same.
```

Comment:
But only it is a local channel update ?

Reply:
If hesitated to add the check, you're right that it would be cleaner, I'll update that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1395/commits/1d0bb60c23f6aa70f970adbaa1ab8358395d55b9